### DISCUSSION TITLE
Installing LAMMPS

### DISCUSSION THREAD

**Discussion by JayKayNJIT** [deepmodeling/deepmd-kit#1702] [General]
Hi, I have successfully gone through the training phase but I am getting stuck at LAMMPS running phase. The first instruction under build-in mode at this page[1], you are saying go to directory "cd $deepmd_source_dir/source/build". But I don't have build directory there, my source directory looks like [2]. 

Thanks for the code.

[1]
https://github.com/deepmodeling/deepmd-kit/blob/master/doc/install/install-lammps.md

[2]
login-1-224 source >: pwd
[...]/DeePMDkit/deepmd-kit-2.1.0/source
login-1-225 source >: ls
3rdparty/  api_cc/  cmake/  CMakeLists.txt  config/  gmx/  install/  ipi/  lib/  lmp/  md/  op/  tests/

**Comment by AnguseZhang** [comment]
Do you have to install DeePMD-kit's C++ interface from scratch? You can refer to https://docs.deepmodeling.com/projects/deepmd/en/master/getting-started/install.html#install-off-line-packages and  https://docs.deepmodeling.com/projects/deepmd/en/master/getting-started/install.html#install-with-conda for an easy installation mode. An executable `lmp` which is patched with DeePMD-kit will also be installed.

**Reply by JayKayNJIT** [reply]
Yes, I installed it by using condo. Please look at my log[1]. I am using lmp binary that comes patched in conda. Please look for my LAMMPS input[2].

Thank you.

[1]
login-1-183 LAMMPS_JOB >: conda activate deepmd
(deepmd) login-1-184 LAMMPS_JOB >: lmp -in lammps.in
LAMMPS (29 Sep 2021 - Update 3)
OMP_NUM_THREADS environment is not set. Defaulting to 1 thread. (src/comm.cpp:98)
  using 1 OpenMP thread(s) per MPI task
2022-05-11 18:00:58.349642: I tensorflow/stream_executor/platform/default/dso_loader.cc:53] Successfully opened dynamic library libcudart.so.11.0
Loading plugin: deepmd pair style v2.0 by Han Wang
Loading plugin: compute deeptensor/atom v2.0 by Han Wang
Loading plugin: fix dplr v2.0 by Han Wang
Reading data file ...
  orthogonal box = (0.0000000 0.0000000 0.0000000) to (31.916280 33.156000 23.204787)
  1 by 1 by 1 MPI processor grid
  reading atoms ...
  267 atoms
  read_data CPU = 0.087 seconds
251 atoms in group membrane
16 atoms in group DNA
Summary of lammps deepmd module ...
  >>> Info of deepmd-kit:
  installed to:       /home/j/jk435/miniconda3/envs/deepmd
  source:             v2.0.3
  source branch:       HEAD
  source commit:      159e45d2
  source commit at:   2021-10-15 10:31:09 +0800
  surpport model ver.:1.0
  build float prec:   double
  build with tf inc:  /home/j/jk435/miniconda3/envs/deepmd/include;/home/j/jk435/miniconda3/envs/deepmd/include
  build with tf lib:  /home/j/jk435/miniconda3/envs/deepmd/lib/libtensorflow_cc.so;/home/j/jk435/miniconda3/envs/deepmd/lib/libtensorflow_framework.so
  set tf intra_op_parallelism_threads: 0
  set tf inter_op_parallelism_threads: 0
  >>> Info of lammps module:
  use deepmd-kit at:  /home/j/jk435/miniconda3/envs/deepmd
  source:             v2.0.3
  source branch:      HEAD
  source commit:      159e45d2
  source commit at:   2021-10-15 10:31:09 +0800
  build float prec:   double
build with tf inc:  /home/j/jk435/miniconda3/envs/deepmd/include;/home/j/jk435/miniconda3/envs/deepmd/include
  build with tf lib:  /home/j/jk435/miniconda3/envs/deepmd/lib/libtensorflow_cc.so;/home/j/jk435/miniconda3/envs/deepmd/lib/libtensorflow_framework.so
2022-05-11 18:01:03.813443: I tensorflow/stream_executor/platform/default/dso_loader.cc:53] Successfully opened dynamic library libcudart.so.11.0
2022-05-11 18:01:03.903627: I tensorflow/core/platform/cpu_feature_guard.cc:142] This TensorFlow binary is optimized with oneAPI Deep Neural Network Library (oneDNN) to use the following CPU instructions in performance-critical operations:  SSE4.1 SSE4.2 AVX
To enable them in other operations, rebuild TensorFlow with the appropriate compiler flags.
2022-05-11 18:01:03.925077: I tensorflow/stream_executor/platform/default/dso_loader.cc:53] Successfully opened dynamic library libcuda.so.1
2022-05-11 18:01:03.998073: E tensorflow/stream_executor/cuda/cuda_driver.cc:328] failed call to cuInit: CUDA_ERROR_NO_DEVICE: no CUDA-capable device is detected
2022-05-11 18:01:03.999204: I tensorflow/stream_executor/cuda/cuda_diagnostics.cc:156] kernel driver does not appear to be running on this host (login-1): /proc/driver/nvidia/version does not exist
2022-05-11 18:01:04.000623: I tensorflow/core/common_runtime/process_util.cc:146] Creating new thread pool with default inter op setting: 2. Tune using inter_op_parallelism_threads for best performance.
2022-05-11 18:01:07.430992: I tensorflow/core/platform/profile_utils/cpu_utils.cc:114] CPU Frequency: 2700000000 Hz
  >>> Info of model(s):
  using   1 model(s): graph-compress.pb
  rcut in model:      6
  ntypes in model:    6
Displacing atoms ...
ERROR: Unrecognized compute style 'deeptensor/atom' (src/modify.cpp:1265)
Last command: compute         dipole all deeptensor/atom graph-compress.pb
(deepmd) login-1-185 LAMMPS_JOB >:



[2]
units           metal
boundary        p p p
atom_style      atomic

read_data        frame0.data
group            membrane type 1 2
group            DNA type 3 4 5 6

mass 1 95.95          # Mo
mass 2 32.065         # S
mass 3 15.999         # O
mass 4 14.0067        # N
mass 5 12.0107        # C
mass 6 1.00784        # H

pair_style deepmd graph-compress.pb
pair_coeff     * *

label loop
variable steps equal round($(lz)/0.5)
variable a loop ${steps}
displace_atoms  DNA move 0 0 0.5

compute         dipole all deeptensor/atom graph-compress.pb
compute ke all ke/atom
compute pe all pe/atom
compute stress all centroid/stress/atom NULL virial
compute flux all heat/flux ke pe stress
run 0
write_dump      all custom frame${a}.data id type x y z c_stress[1] c_stress[2] c_stress[3]
next a
jump SELF loop
print "ALL DONE"

**Reply by njzjz** [reply]
> ERROR: Unrecognized compute style 'deeptensor/atom' (src/modify.cpp:1265)

It has been fixed by #1548 (released in v2.1.1).

**Reply by JayKayNJIT** [reply]
The installation is exiting with below error:

login-1-46 DeePMD_2.1.1 >: conda create -n deepmd deepmd-kit=2.1.1=*cpu libdeepmd=2.1.1=*cpu lammps-dp=2.1.1 horovod -c https://conda.deepmodeling.org
Collecting package metadata (current_repodata.json): done
Solving environment: failed with repodata from current_repodata.json, will retry with next repodata source.
Collecting package metadata (repodata.json): done
Solving environment: failed

PackagesNotFoundError: The following packages are not available from current channels:

  - lammps-dp=2.1.1

Current channels:

  - https://conda.deepmodeling.org/linux-64
  - https://conda.deepmodeling.org/noarch
  - https://repo.anaconda.com/pkgs/main/linux-64
  - https://repo.anaconda.com/pkgs/main/noarch
  - https://repo.anaconda.com/pkgs/r/linux-64
  - https://repo.anaconda.com/pkgs/r/noarch

To search for alternate channels that may provide the conda package you're
looking for, navigate to

    https://anaconda.org

and use the search bar at the top of the page.


login-1-46 DeePMD_2.1.1 >:

**Reply by njzjz** [reply]
Remove the version constrict of `lammps-dp`:
```
conda create -n deepmd deepmd-kit=2.1.1=*cpu libdeepmd=2.1.1=*cpu lammps-dp horovod -c https://conda.deepmodeling.org/
```

**Reply by JayKayNJIT** [reply]
Thanks, installation worked fine. But the LAMMPS job is exiting with an error[1], and log file[2].

[1]
login-1-53 LAMMPS_JOB >: conda activate deepmd
(deepmd) login-1-54 LAMMPS_JOB >: ls
frame0.data  graph-compress.pb  lammps.in  log.lammps
(deepmd) login-1-55 LAMMPS_JOB >: lmp -in lammps.in
LAMMPS (29 Sep 2021 - Update 3)
OMP_NUM_THREADS environment is not set. Defaulting to 1 thread. (src/comm.cpp:98)
  using 1 OpenMP thread(s) per MPI task
Loading plugin: deepmd pair style v2.0 by Han Wang
Loading plugin: compute deeptensor/atom v2.0 by Han Wang
Loading plugin: fix dplr v2.0 by Han Wang
Reading data file ...
  orthogonal box = (0.0000000 0.0000000 0.0000000) to (31.916280 33.156000 23.204787)
  1 by 1 by 1 MPI processor grid
  reading atoms ...
  267 atoms
  read_data CPU = 0.002 seconds
251 atoms in group membrane
16 atoms in group DNA
Summary of lammps deepmd module ...
  >>> Info of deepmd-kit:
  installed to:       /home/j/jk435/miniconda3/envs/deepmd
  source:             v2.1.1
  source branch:       HEAD
  source commit:      c4f0cec0
  source commit at:   2022-04-16 11:11:16 +0800
  surpport model ver.:1.1
  build float prec:   double
  build with tf inc:  /home/j/jk435/miniconda3/envs/deepmd/include;/home/j/jk435/miniconda3/envs/deepmd/include
  build with tf lib:  /home/j/jk435/miniconda3/envs/deepmd/lib/libtensorflow_cc.so;/home/j/jk435/miniconda3/envs/deepmd/lib/libtensorflow_framework.so
  set tf intra_op_parallelism_threads: 0
  set tf inter_op_parallelism_threads: 0
  >>> Info of lammps module:
  use deepmd-kit at:  /home/j/jk435/miniconda3/envs/deepmd2022-05-16 14:42:27.484085: I tensorflow/core/platform/cpu_feature_guard.cc:151] This TensorFlow binary is optimized with oneAPI Deep Neural Network Library (oneDNN) to use the following CPU instructions in performance-critical operations:  SSE4.1 SSE4.2 AVX
To enable them in other operations, rebuild TensorFlow with the appropriate compiler flags.
2022-05-16 14:42:27.500263: I tensorflow/core/common_runtime/process_util.cc:146] Creating new thread pool with default inter op setting: 2. Tune using inter_op_parallelism_threads for best performance.
 >>> Info of model(s):
  using   1 model(s): graph-compress.pb
  rcut in model:      6
  ntypes in model:    6
Displacing atoms ...
Summary of lammps deepmd module ...
  >>> Info of deepmd-kit:
  installed to:       /home/j/jk435/miniconda3/envs/deepmd
  source:             v2.1.1
  source branch:       HEAD
  source commit:      c4f0cec0
  source commit at:   2022-04-16 11:11:16 +0800
  surpport model ver.:1.1
  build float prec:   double
  build with tf inc:  /home/j/jk435/miniconda3/envs/deepmd/include;/home/j/jk435/miniconda3/envs/deepmd/include
  build with tf lib:  /home/j/jk435/miniconda3/envs/deepmd/lib/libtensorflow_cc.so;/home/j/jk435/miniconda3/envs/deepmd/lib/libtensorflow_framework.so
  set tf intra_op_parallelism_threads: 0
  set tf inter_op_parallelism_threads: 0
  >>> Info of lammps module:
  use deepmd-kit at:  /home/j/jk435/miniconda3/envs/deepmdINVALID_ARGUMENT: Tensor model_attr/output_dim:0, specified in either feed_devices or fetch_devices was not found in the Graph
terminate called after throwing an instance of 'deepmd::tf_exception'
  what():  DeePMD-kit Error: TensorFlow Error: INVALID_ARGUMENT: Tensor model_attr/output_dim:0, specified in either feed_devices or fetch_devices was not found in the Graph
Aborted
(deepmd) login-1-56 LAMMPS_JOB >: ls
frame0.data  graph-compress.pb  lammps.in  log.lammps

[2]
LAMMPS (29 Sep 2021 - Update 3)
OMP_NUM_THREADS environment is not set. Defaulting to 1 thread. (src/comm.cpp:98)
  using 1 OpenMP thread(s) per MPI task
plugin load libdeepmd_lmp.so
Loading plugin: deepmd pair style v2.0 by Han Wang
Loading plugin: compute deeptensor/atom v2.0 by Han Wang
Loading plugin: fix dplr v2.0 by Han Wang
units           metal
boundary        p p p
atom_style      atomic

#read_data       23_12.data # 1st Mo, 2nd S
#read_data       DFTrelaxed.data extra/atom/types 4 group membrane
#read_data       Conformer3D_CID_135398634_Guanine.data offset 2 0 0 0 0 add append group DNA
read_data        frame0.data
Reading data file ...
  orthogonal box = (0.0000000 0.0000000 0.0000000) to (31.916280 33.156000 23.204787)
  1 by 1 by 1 MPI processor grid
  reading atoms ...
  267 atoms
  read_data CPU = 0.002 seconds
group            membrane type 1 2
251 atoms in group membrane
group            DNA type 3 4 5 6
16 atoms in group DNA

mass 1 95.95          # Mo
mass 2 32.065         # S
mass 3 15.999         # O
mass 4 14.0067        # N
mass 5 12.0107        # C
mass 6 1.00784        # H

pair_style deepmd graph-compress.pb
Summary of lammps deepmd module ...
  >>> Info of deepmd-kit:
  installed to:       /home/j/jk435/miniconda3/envs/deepmd
  source:             v2.1.1
  source branch:       HEAD
  source commit:      c4f0cec0
  source commit at:   2022-04-16 11:11:16 +0800
  surpport model ver.:1.1
  build float prec:   double
  build with tf inc:  /home/j/jk435/miniconda3/envs/deepmd/include;/home/j/jk435/miniconda3/envs/deepmd/include
  build with tf lib:  /home/j/jk435/miniconda3/envs/deepmd/lib/libtensorflow_cc.so;/home/j/jk435/miniconda3/envs/deepmd/lib/libtensorflow_framework.so
  set tf intra_op_parallelism_threads: 0
  set tf inter_op_parallelism_threads: 0
 >>> Info of lammps module:
  use deepmd-kit at:  /home/j/jk435/miniconda3/envs/deepmdpair_coeff     * *

#displacing DNA Nucleobases and generating frames
#write_dump       all cfg frame0.data mass type xs ys zs modify element Mo S O N C H
#write_dump      all custom frame0.data id type x y z
label loop
variable steps equal round($(lz)/0.5)
variable steps equal round(23.204787034399998902/0.5)
variable a loop ${steps}
variable a loop 46
displace_atoms  DNA move 0 0 0.5
Displacing atoms ...
#write_dump       all cfg frame${a}.data mass type xs ys zs modify element Mo S O N C H
compute         dipole all deeptensor/atom graph-compress.pb
Summary of lammps deepmd module ...
  >>> Info of deepmd-kit:
  installed to:       /home/j/jk435/miniconda3/envs/deepmd
  source:             v2.1.1
  source branch:       HEAD
  source commit:      c4f0cec0
  source commit at:   2022-04-16 11:11:16 +0800
  surpport model ver.:1.1
  build float prec:   double
  build with tf inc:  /home/j/jk435/miniconda3/envs/deepmd/include;/home/j/jk435/miniconda3/envs/deepmd/include
  build with tf lib:  /home/j/jk435/miniconda3/envs/deepmd/lib/libtensorflow_cc.so;/home/j/jk435/miniconda3/envs/deepmd/lib/libtensorflow_framework.so
  set tf intra_op_parallelism_threads: 0
  set tf inter_op_parallelism_threads: 0
  >>> Info of lammps module:
  use deepmd-kit at:  /home/j/jk435/miniconda3/envs/deepmd

**Reply by njzjz** [reply]
I don't find any error message..

**Reply by JayKayNJIT** [reply]
I am sorry, I edited the original post which includes the error message now. Thanks.

**Reply by njzjz** [reply]
>  compute dipole all deeptensor/atom graph-compress.pb

Looks like you are trying to load an energy model to compute the dipole. Instead, you need to train a dipole model. There is [an example](https://github.com/deepmodeling/deepmd-kit/tree/v2.1.1/examples/water_tensor/dipole). See also the [doc](https://docs.deepmodeling.com/projects/deepmd/en/v2.1.1/model/train-fitting-tensor.html).

**Reply by JayKayNJIT** [reply]
Perfect, it's working now. Thanks a lot!
